src/middlewared/middlewared/plugins/crypto_/certificate_authorities.py

```diff
@@ -126,33 +127,16 @@ async def get_serial_for_certificate(self, ca_id):
 
             async def cert_serials(ca_id):
                 return [
-                    data['serial'] for data in
-                    await self.middleware.call(
-                        'datastore.query',
-                        'system.certificate',
-                        [('signedby', '=', ca_id)],
-                        {
-                            'prefix': self._config.datastore_prefix,
-                            'extend': self._config.datastore_extend
-                        }
+                    data['serial'] for data in await self.middleware.call(
```

Comment:
Why do we switch to `certificate.query` if we only need serials?

Reply:
We need to parse the certificate to get the serial and on that note we don't need all the other normalization, i'll quickly make changes to do just that